src/operations/SymbolicOperation.cpp

```diff
@@ -332,4 +332,54 @@ void SymbolicOperation::instantiate(const VariableAssignment& assignment) {
   }
   checkUgate();
 }
+
+void SymbolicOperation::invert() {
+  auto negate = [this](const size_t index) {
+    if (isSymbolicParameter(index)) {
+      symbolicParameter.at(index) = -symbolicParameter.at(index).value();
+    } else {
+      parameter.at(index) = -parameter.at(index);
+    }
+  };
+  auto add = [this](const size_t index, const fp value) {
+    if (isSymbolicParameter(index)) {
+      symbolicParameter.at(index) = symbolicParameter.at(index).value() + value;
+    } else {
+      parameter.at(index) += value;
+    }
+  };
+
+  switch (type) {
+  case GPhase:
+  case Phase:
+  case RX:
+  case RY:
+  case RZ:
+  case RXX:
+  case RYY:
+  case RZZ:
+  case RZX:
+    negate(0);
+    break;
+  case U2:
+    negate(0);
+    negate(1);
+
+    add(0, -PI);
+    add(1, PI);
+    std::swap(parameter[0], parameter[1]);
+    std::swap(symbolicParameter[0], symbolicParameter[1]);
+    break;
+  case U3:
+    negate(0);
+    negate(1);
+    negate(2);
+
+    std::swap(parameter[1], parameter[2]);
+    std::swap(symbolicParameter[1], symbolicParameter[2]);
+    break;
```

Comment:
just for safety: also make sure to cover all uncovered operations (with parameters) from the `StandardOperation` file.